From 45560c8abee000b3f0e5fcedd2e34ca8e2039309 Mon Sep 17 00:00:00 2001
From: Francis De Brabandere <francisdb@gmail.com>
Date: Fri, 17 Apr 2026 22:11:41 +0200
Subject: [PATCH 5/9] vbscript: Include function name and line in resume-next

---
 dlls/vbscript/interp.c | 7 ++++++-
 1 file changed, 6 insertions(+), 1 deletion(-)

diff --git a/dlls/vbscript/interp.c b/dlls/vbscript/interp.c
index 1bbcd7e0000..72e87a7c684 100644
--- a/dlls/vbscript/interp.c
+++ b/dlls/vbscript/interp.c
@@ -2896,8 +2896,13 @@ HRESULT exec_script(script_ctx_t *ctx, BOOL extern_caller, function_t *func, vbd
 
             if(exec.resume_next) {
                 unsigned stack_off;
+                unsigned line = exec.code->start_line + 1;
+                const WCHAR *nl;
+                for(nl = exec.code->source; nl < exec.code->source + exec.instr->loc; nl++)
+                    if(*nl == '\n') line++;
 
-                WARN("Failed %08lx in resume next mode\n", ctx->ei.scode);
+                WARN("Failed %08lx in resume next mode in %s, line %u\n", ctx->ei.scode,
+                     exec.func->name ? debugstr_w(exec.func->name) : "<global>", line);
 
                 /*
                  * Unwinding here is simple. We need to find the next OP_catch, which contains
-- 
2.54.0

