From 017e0e70beaa8ef948fabcf1aa6759ab88bd201b Mon Sep 17 00:00:00 2001
From: Francis De Brabandere <francisdb@gmail.com>
Date: Thu, 16 Apr 2026 14:09:00 +0200
Subject: [PATCH 06/11] test: Add Noop builtin for variadic stubbing of VPX

---
 dlls/vbscript/global.c | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/dlls/vbscript/global.c b/dlls/vbscript/global.c
index ba9cb52fd64..d16237fc6da 100644
--- a/dlls/vbscript/global.c
+++ b/dlls/vbscript/global.c
@@ -3863,6 +3863,11 @@ static HRESULT Global_GetRef(BuiltinDisp *This, VARIANT *arg, unsigned args_cnt,
     return MAKE_VBSERROR(VBSE_ILLEGAL_FUNC_CALL);
 }
 
+static HRESULT Global_BuiltinNoop(BuiltinDisp *This, VARIANT *args, unsigned args_cnt, VARIANT *res)
+{
+    return S_OK;
+}
+
 static HRESULT Global_CreateCollection(BuiltinDisp *This, VARIANT *args, unsigned args_cnt, VARIANT *res)
 {
     IDispatch *disp;
@@ -4048,6 +4053,7 @@ static const builtin_prop_t global_props[] = {
     {L"Month",                     Global_Month, 0, 1},
     {L"MonthName",                 Global_MonthName, 0, 1, 2},
     {L"MsgBox",                    Global_MsgBox, 0, 1, 5},
+    {L"Noop",                      Global_BuiltinNoop, 0, 0, MAXDWORD},
     {L"Now",                       Global_Now, 0, 0},
     {L"Oct",                       Global_Oct, 0, 1},
     {L"Randomize",                 Global_Randomize, 0, 0, 1},
-- 
2.53.0

